dist/jszip-utils.js

```diff
@@ -49,7 +49,18 @@ var createXHR = window.ActiveXObject ?
 
 
 
-JSZipUtils.getBinaryContent = function(path, callback) {
+JSZipUtils.getBinaryContent = function(path, cbo) {
+    //old callback style
+    if(typeof cbo==="function"){
+        var callback = cbo;
+        cbo = {
+            done: function(data){ callback(null, data); },
+            fail: function(err){ callback(err, null); }
+        };
+    }
```

Comment:
Given the comment below about the options properties, I think this should be changed to

```js
    if (typeof options === "function") {
        options = {callback: options};
    }
```

Reply:
Do you think I should return a promise only when no callback is provided? I think that way we don't even try to create a new Promise() on legacy codebases, so it stays just like before.

about the options should it be `(path, callback)` and to use options you `(path, {callback})` or it can  be `(path, callback, options)`

and `(path, options)` for promises, I've seen this being use on node like [here](https://github.com/jprichardson/node-fs-extra/blob/master/lib/copy/copy.js)

the tradeoff is that autocomplete gets a bit confused by the offsite parameters

although I have no preference here, just wanted to bring it up.

Reply:
I followed the same standard as some node libs, that when a callback is provided, it doesn't return a promise.